### PR TITLE
AST-2280 -  Narrow Width Container Layout

### PR DESCRIPTION
**Pull request by neerajmasai**
### Description

- Feature: New Narrow Width Container Layout:
                 -> Global.
                 -> Narrow width slider controller.
                 -> Blogs: Single/Archives.
                 -> Pages.
                 -> Page Meta.
                 -> Block Editor Compatibility.
### Screenshots

- Feature: New Narrow Width Container Layout:
                 -> Global. - https://d.pr/i/TQfIju
                 -> Narrow width slider controller. - https://d.pr/i/TQfIju
                 -> Blogs: Single/Archives. - https://d.pr/i/sbvkvu / https://d.pr/i/0Yoy0w
                 -> Pages. - https://d.pr/i/XS2Q0y
                 -> Page Meta. - https://d.pr/i/e4fGrt
                 -> Block Editor Compatibility.
### Types of changes
- Implemented configs for blogs, pages, page meta, global.
- New svg for narrow container width.
- New narrow container width slider to adjust width for container.
- No sidebar displayed in case of narrow width container.
- Customizer preview changes.

- Feature: New Narrow Width Container Layout:
                 -> Global.
                 -> Narrow width slider controller.
                 -> Blogs: Single/Archives.
                 -> Pages.
                 -> Page Meta.
                 -> Block Editor Compatibility.

### How has this been tested?

Testcases:

1. When we select NWL from Global - should apply globally, unless overidden by Blogs / Pages settings to a diff layout.
2. When we select NWL from Blog Archive - should apply NWL to archives or if selected as default & global is NWL. Overriden by Page Meta.
3. When we select NWL from Blog Single - should apply NWL to single posts or if selected as default & global is NWL. Overriden by Page Meta.
4. When we select NWL from Pages - should apply NWL to static pages or if selected as default & global is NWL. Overriden by Page Meta.
5. Sidebar - if we select NWL in any of the above areas, Sidebar should not be visible.
6. Page Meta - if we select NWL from page meta, then NWL should apply with highest priority or if default & global is NWL.
7. Narrow Width Slider - width of the NWL should be adjustable & will apply to all layouts, default 750px.
8. Block Editor - In editor, NWL will have the narrow width, can be adjusted from Customizer through slider, check different blocks & they should look same in frontend, just like other layouts.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by yssambare12 — APPROVED** [review]

**Review by nathanrodrigues2111 — COMMENTED** [review]
Please add changelogs

File: inc/class-astra-dynamic-css.php [hunk 1]
```diff
@@ -62,8 +62,12 @@ public static function return_output( $dynamic_css, $dynamic_css_filtered = '' )
 			 */
 			$is_site_rtl                = is_rtl();
 			$site_content_width         = astra_get_option( 'site-content-width', 1200 );
+			$narrow_container_max_width = astra_get_option( 'narrow-container-max-width', 750 );
```
Comment (imnavanath) [inline inc/class-astra-dynamic-css.php:65]:
```suggestion
			$narrow_container_max_width = astra_get_option( 'narrow-container-max-width', apply_filters( 'astra_narrow_container_width', 750 ) );
```

Comment (imnavanath) [reply inc/class-astra-dynamic-css.php:65]: Let's have a flexibility to extend this option.

File: inc/core/class-astra-wp-editor-css.php [hunk 2]
```diff
@@ -386,6 +388,11 @@ public static function get_css() {
 			--wp--custom--ast-wide-width-size: ' . $ast_wide_width . ';
 		}';
 
+		$css .= ':root, .ast-narrow-container .editor-styles-wrapper {
```
Comment (imnavanath) [inline inc/core/class-astra-wp-editor-css.php:391]:
```suggestion
		$css .= '.ast-narrow-container .editor-styles-wrapper {
```

Comment (imnavanath) [reply inc/core/class-astra-wp-editor-css.php:391]: Please verify once

File: inc/core/common-functions.php [hunk 3]
```diff
@@ -1620,3 +1620,38 @@ function astra_load_woocommerce_login_form_password_icon() {
...
 	return apply_filters( 'astra_get_option_woo-show-password-icon', isset( $astra_settings['woo-show-password-icon'] ) ? false : true ); // phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
 }
+
+/**
+ * Function to add narrow width properties in the frontend.
+ *
+ * @since x.x.x
+ * @param string  $location container layout for single-post, archives, pages, page meta.
+ * @param string $narrow_container_max_width  dynamic container width in px.
+ * @return string|null css based on $location and $narrow_container_max_width.
```
Comment (imnavanath) [inline inc/core/common-functions.php:1630]:
```suggestion
 * @return string Parsed CSS based on $location and $narrow_container_max_width.
```

File: inc/core/common-functions.php [hunk 4]
```diff
@@ -1620,3 +1620,38 @@ function astra_load_woocommerce_login_form_password_icon() {
...
+		if ( 'narrow-container' === astra_get_content_layout() ) {
+			add_filter(
+				'astra_page_layout',
+				function() { // phpcs:ignore PHPCompatibility.FunctionDeclarations.NewClosure.Found
+					return 'no-sidebar';
+				}
+			);
+		}
+
+		return astra_parse_css( $narrow_container_css, astra_get_tablet_breakpoint( '', 1 ) ); // phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
```
Comment (imnavanath) [inline inc/core/common-functions.php:1652]: This phpcs:ignore don't need I guess, can you verify once @neerajmasai .

Comment (neerajmasai) [reply inc/core/common-functions.php:1652]: Okay

**Review by imnavanath — APPROVED** [review]

**Review by nathanrodrigues2111 — APPROVED** [review]

File: inc/core/class-astra-wp-editor-css.php [hunk 5]
```diff
@@ -557,6 +562,28 @@ public static function get_css() {
 			'padding-right' => '3px',
 		);
 
+		// Core / Spectra blocks compatibility to occupy same width as narrow container.
+		$desktop_css['.ast-narrow-container .editor-styles-wrapper .block-editor-block-list__layout.is-root-container > *.wp-block, .ast-narrow-container .is-root-container > .alignfull > :where(:not(.alignleft):not(.alignright)), .ast-narrow-container .editor-styles-wrapper .is-root-container > .wp-block-cover.alignfull .wp-block-cover__image-background'] = array(
```
Comment (imnavanath) [inline inc/core/class-astra-wp-editor-css.php:566]: Is this selector necessary - `.wp-block-cover__image-background` seems to be dynamic?? can we remove this.

File: inc/core/class-astra-wp-editor-css.php [hunk 6]
```diff
@@ -557,6 +562,28 @@ public static function get_css() {
...
+			'max-width' => 'var(--wp--custom--ast-content-width-size)',
+			'margin-left' => 'auto',
+			'margin-right' => 'auto',
+		);
+		$desktop_css['.ast-narrow-container .is-root-container > .alignwide > :where(:not(.alignleft):not(.alignright)), .ast-narrow-container .editor-styles-wrapper .is-root-container > .wp-block-cover.alignwide .wp-block-cover__image-background'] = array(
```
Comment (imnavanath) [inline inc/core/class-astra-wp-editor-css.php:571]: Is this selector necessary - `.wp-block-cover__image-background` seems to be dynamic?? can we remove this.

**Pull request merged**